app/src/main/java/com/duckduckgo/app/browser/trafficquality/AndroidFeaturesHeader.kt

```diff
@@ -26,22 +28,27 @@ import javax.inject.Inject
 class AndroidFeaturesHeaderPlugin @Inject constructor(
     private val duckDuckGoUrlDetector: DuckDuckGoUrlDetector,
     private val androidBrowserConfigFeature: AndroidBrowserConfigFeature,
+    private val androidFeaturesHeaderProvider: AndroidFeaturesHeaderProvider,
 ) : CustomHeadersPlugin {
 
     override fun getHeaders(url: String): Map<String, String> {
         if (androidBrowserConfigFeature.self().isEnabled() &&
             androidBrowserConfigFeature.featuresRequestHeader().isEnabled() &&
             duckDuckGoUrlDetector.isDuckDuckGoQueryUrl(url)
         ) {
-            return mapOf(
-                X_DUCKDUCKGO_ANDROID_HEADER to TEST_VALUE,
-            )
+            val headerValue = androidFeaturesHeaderProvider.provide()
+            return if (headerValue != null) {
+                mapOf(
+                    X_DUCKDUCKGO_ANDROID_HEADER to headerValue,
+                )
+            } else {
+                emptyMap()
+            }
```

Comment:
This can be simplified:

```suggestion
            androidFeaturesHeaderProvider.provide()?.let { headerValue ->
                return mapOf(X_DUCKDUCKGO_ANDROID_HEADER to headerValue)
            }
```